### ISSUE TITLE
Replace QtAwesome usage by direct SVG icons

### ISSUE THREAD

**Issue by goanpeca** [spyder-ide/spyder#4064]
Since the issues present on Windows10 with loading fonts (which qt awesome relies on) we need to move to a better system. Additionally I think we really need to design icons from the ground up and not just use FontAwesome, which by today standards are a bit outdated. There is an additional burden in creating icons that are based on a font as we have to follow font metrics (bearings etc...).

This issue is related to the bigger issues of having themes in spyder that allow to control not only the appearance of the color schemes in the editor, but the whole application.

The first step is to use the current svg icons (export all the one in use) and define base SVG from which we can then generate all the other states. Svg are text files so we can read them and adapt colors based on replace tags (using string `{format}`). The base files should be shipped with spyder and the themes will be generated on the fly and stored in the configuration folder.

`.spyder/themes/theme-name/files.svg ...`

https://github.com/ChrisRM/material-theme-jetbrains

https://github.com/equinusocio/material-theme

**Comment by flying-sheep** [comment]
hi! that’s pretty much exactly what i initially proposed in #2230, but nobody believed me then 😜

there’s also another issue, i.e. that monochrome icons don’t work well if the background happens to be similar to the single color.

maybe since i was right, you’re now more open to my proposals:

1. #2230: we can use `QIcon.fromTheme` with a fallback theme loaded into spyder’s `.qrc` file. (as implemented in takluyver/nbmanager#10)

2. we can use and expand KDE’s breeze icon theme as that default theme:

    - it has many preexisting icons (so we’ll find a matching one for many of the icons we need)
    - it’s simple and has good style guideline specs, so we can easily create fitting icons that don’t exist yet
    - it fits the flat aesthetic you were going for with FontAwesome
    - it has a [system](http://notmart.org/blog/2016/05/icon-colors/) for adapting the primary color to the text color (i.e. it works on any background)

**Comment by ccordoba12** [comment]
I'm closing this one as a duplicate of issue #2350 because when we fix that one, we'll fix this.

**Comment by nsgates** [comment]
It looks like this issue has not been fixed despite the dark theme implementation. Is it still on the roadmap somewhere? The problem of FontAwesome (etc.) being blocked on Windows is difficult to overcome in managed computing environments. In addition, changing the preferences to use the Spyder 3 icon theme causes Spyder to crash on launch, which requires manually editing (and in many cases generating) the spyder.ini file in order to overcome. Using the Spyder 2 icon theme leads to many missing icons (replaced with red X's) in Spyder 5. Are there any plans to deal with the issue of Windows blocking FontAwesome and any other dependencies of Qtawesome?

**Comment by ccordoba12** [comment]
> Is it still on the roadmap somewhere?

I'm sorry but we ruled out that possibility because it's much easier for us to depend on the icons provided by QtAwesome.

> Are there any plans to deal with the issue of Windows blocking FontAwesome? (and any other dependencies of Qtawesome?)

We've talked about installing the QtAwesome fonts locally so that users don't encounter permissions problems due to that. @dalthviz, what do you think about this? Could you work on it on the second half of the year?

**Comment by ccordoba12** [comment]
Great! Thanks for taking care of that.
